### PR TITLE
[PF-2626] Include flex resources in workspace clone flight.

### PR DESCRIPTION
**Pull request by aaron-wang**
All the cloning logic is done from #1133 . We just need to put it in the clone workspace flight.

### CODE REVIEW

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/flexibleresource/CloneControlledFlexibleResourceFlight.java [hunk 1]
```diff
@@ -42,7 +42,6 @@ public CloneControlledFlexibleResourceFlight(FlightMap inputParameters, Object b
         inputParameters,
         WorkspaceFlightMapKeys.ResourceKeys.RESOURCE,
         JobMapKeys.AUTH_USER_INFO.getKeyName(),
-        WorkspaceFlightMapKeys.ResourceKeys.CLONING_INSTRUCTIONS,
```
Comment (aaron-wang) [inline service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/flexibleresource/CloneControlledFlexibleResourceFlight.java:45]: Removed this unnecessary restriction: The resolved cloning instructions are calculated later regardless.

If the cloning instructions are not provided in the input parameters, then we use the source flex resource cloning instructions.

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/LaunchCloneControlledFlexibleResourceFlightStep.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import bio.terra.stairway.exception.StairwayExecutionException;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.job.JobMapKeys;
+import bio.terra.workspace.service.resource.controlled.cloud.any.flexibleresource.ControlledFlexibleResource;
+import bio.terra.workspace.service.resource.controlled.flight.clone.flexibleresource.CloneControlledFlexibleResourceFlight;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import java.util.UUID;
+import javax.annotation.Nullable;
+
+public class LaunchCloneControlledFlexibleResourceFlightStep implements Step {
```
Comment (aaron-wang) [inline service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/LaunchCloneControlledFlexibleResourceFlightStep.java:22]: So, I tested this locally with a few cases via Swagger. 

Most importantly: I had flex resource A with `COPY_RESOURCE`, and flex resource B with `COPY_NOTHING`. 

Resource A copied over, and resource B did not copy over (thus, the correct behavior).

Do we need to add tests confirming the flex resources are cloned over during the workspace clone flight?

Comment (ddietterich) [reply service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/LaunchCloneControlledFlexibleResourceFlightStep.java:22]: You already validate the resource clone operation. So the additional piece to cover is making sure that the resources are included in the clone. Perhaps add a flex resource to the integration `CloneWorkspace` test. I think the test structure is pretty obvious.

**Review by ddietterich — APPROVED** [review]
LGTM - I would suggest extending the CloneWorkspace integration test to add a flex resource.

**Review by zloery — APPROVED** [review]

**Pull request merged**